luigi/worker.py

```diff
@@ -678,6 +707,26 @@ def _log_remote_tasks(self, running_tasks, n_pending_tasks, n_unique_pending):
             if n_unique_pending:
                 logger.debug("There are %i pending tasks unique to this worker", n_unique_pending)
 
+    def _get_work_task_id(self, get_work_response):
+        if get_work_response['task_id'] is not None:
+            return get_work_response['task_id']
+        elif 'batch_id' in get_work_response:
+            task = load_task(
+                module=get_work_response.get('task_module'),
+                task_name=get_work_response['task_family'],
+                params_str=get_work_response['task_params'],
+            )
+            self._scheduler.add_task(
+                worker=self._id,
+                task_id=task.task_id,
+                module=get_work_response.get('task_module'),
+                family=get_work_response['task_family'],
+                params=task.to_str_params(),
+                status=RUNNING,
+                batch_id=get_work_response['batch_id'],
+            )
```

Comment:
What you say makes sense. I just found the code a bit hard to read, in particular the `params_str=get_work_response['task_params']` still looks like we're talking about a single task, but `get_work_response['task_params']` is in fact a dict of lists now. 

-------

It's fine to keep this I think. I don't know any reason for us to keep the RUNNING invariant,